docs/table-migration.md

```diff
@@ -0,0 +1,129 @@
+---
+title: "Table Migration"
+url: table-migration
+weight: 400
+menu: main
```

Comment:
Since this is a new page, I would like to put a preview of the new page organization here:
![Screen Shot 2023-03-22 at 20 08 00](https://user-images.githubusercontent.com/55990951/227065989-003e075a-9ca6-46ec-913e-b62034cef355.png)

Currently I made the `Table Migration` in the main menu.
The preview is built via instructions in https://github.com/apache/iceberg-docs

Reply:
I think we are missing some introduction of the general idea of migration. There should be 2 cases, where (1) users want to migrate using something like a CTAS, that is actually preferred because you completely cut tie with your old table. But it takes more time and might not be possible for production use cases. (2) users use in place migration, which retains existing files but add Iceberg metadata on top of it. And what we have in this page explains more about the later approach. Then we can divide into Hive and Delta cases to explain further.